ibis/pyspark/compiler.py

```diff
@@ -1011,3 +1058,309 @@ def compile_cumulative_max(t, expr, scope, *, window, **kwargs):
+    6: (4, 'Friday'),
+    7: (5, 'Saturday')
+}
+
+
+def _get_ibis_day_of_week_index(pyspark_index):
+    return _day_of_week_mapping[pyspark_index][0]
+
+
+def _get_day_of_week_name(pyspark_index):
+    return _day_of_week_mapping[pyspark_index][1]
+
+
+@compiles(ops.Date)
+def compile_date(t, expr, scope, **kwargs):
+    op = expr.op()
+    src_column = t.translate(op.arg, scope)
+    return F.to_date(src_column).cast('timestamp')
+
+
+def _extract_x_from_datetime(t, expr, scope, extract_fn, **kwargs):
+    op = expr.op()
+    date_col = t.translate(op.arg, scope)
+    return extract_fn(date_col)
+
+
+@compiles(ops.ExtractYear)
+def compile_extract_year(t, expr, scope, **kwargs):
+    return _extract_x_from_datetime(t, expr, scope, F.year, **kwargs)
+
+
+@compiles(ops.ExtractMonth)
+def compile_extract_month(t, expr, scope, **kwargs):
+    return _extract_x_from_datetime(t, expr, scope, F.month, **kwargs)
+
+
+@compiles(ops.ExtractDay)
+def compile_extract_day(t, expr, scope, **kwargs):
+    return _extract_x_from_datetime(t, expr, scope, F.dayofmonth, **kwargs)
+
+
+@compiles(ops.ExtractHour)
+def compile_extract_hour(t, expr, scope, **kwargs):
+    return _extract_x_from_datetime(t, expr, scope, F.hour, **kwargs)
+
+
+@compiles(ops.ExtractMinute)
+def compile_extract_minute(t, expr, scope, **kwargs):
+    return _extract_x_from_datetime(t, expr, scope, F.minute, **kwargs)
+
+
+@compiles(ops.ExtractSecond)
+def compile_extract_second(t, expr, scope, **kwargs):
+    return _extract_x_from_datetime(t, expr, scope, F.second, **kwargs)
+
+
+@compiles(ops.ExtractMillisecond)
+def compile_extract_millisecond(t, expr, scope, **kwargs):
+    raise com.UnsupportedOperationError(
+        'PySpark backend does not support extracting milliseconds.'
+    )
+
+
+@compiles(ops.DateTruncate)
+def compile_date_truncate(t, expr, scope, **kwargs):
+    op = expr.op()
+
+    try:
+        unit = _time_unit_mapping[op.unit]
+    except KeyError:
+        raise com.UnsupportedOperationError(
+            '{!r} unit is not supported in timestamp truncate'.format(op.unit)
+        )
+
+    src_column = t.translate(op.arg, scope)
+    return F.date_trunc(unit, src_column)
+
+
+@compiles(ops.TimestampTruncate)
+def compile_timestamp_truncate(t, expr, scope, **kwargs):
+    return compile_date_truncate(t, expr, scope, **kwargs)
+
+
+@compiles(ops.Strftime)
+def compile_strftime(t, expr, scope, **kwargs):
+    raise com.UnsupportedOperationError(
+        'PySpark uses Java SimpleDateFormat and does not support strftime.'
+    )
+
+
+@compiles(ops.TimestampFromUNIX)
+def compile_timestamp_from_unix(t, expr, scope, **kwargs):
+    op = expr.op()
+    unixtime = t.translate(op.arg, scope)
+    if not op.unit:
+        return F.to_timestamp(F.from_unixtime(unixtime))
+    elif op.unit == 's':
+        fmt = 'yyyy-MM-dd HH:mm:ss'
+        return F.to_timestamp(F.from_unixtime(unixtime, fmt), fmt)
+    else:
+        raise com.UnsupportedArgumentError(
```

Comment:
What is the use case that would hit this?

Reply:
The unsupported argument error? For all units that are not the default none or `s`.

Reply:
So units like `ms`, `us`, `ns`.

Reply:
I thought `ms` should be supported but `F.from_unixtime(unixtime, fmt)` in PySpark does not handle format `yyyy-MM-dd HH:mm:ss.SSS`.

Reply:
Sorry, what I meant is: What user code will trigger this unsupported unit?

Reply:
```
expr = int_expr.to_timestamp(unit)
result = con.execute(expr)
```

where unit is `ms`, `us`, or `ns`.

This is tested in `test_to_timestamp()` in `test_temporal.py`.